contracts/0.4.24/utils/Versioned.sol

```diff
@@ -0,0 +1,39 @@
+// SPDX-FileCopyrightText: 2022 Lido <info@lido.fi>
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity 0.4.24;
+
+import "@aragon/os/contracts/common/UnstructuredStorage.sol";
+
+
+contract Versioned {
+    using UnstructuredStorage for bytes32;
+
+    event ContractVersionSet(uint256 version);
+
+    /// @dev Storage slot: uint256 version
+    /// Version of the initialized contract storage.
+    /// The version stored in CONTRACT_VERSION_POSITION equals to:
+    /// - 0 right after the deployment, before an initializer is invoked (and only at that moment);
+    /// - N after calling initialize(), where N is the initially deployed contract version;
+    /// - N after upgrading contract by calling finalizeUpgrade_vN().
+    bytes32 internal constant CONTRACT_VERSION_POSITION = keccak256("lido.Versioned.contractVersion");
+
+    constructor() {
+        // lock version in the implementation's storage to prevent initialization
+        CONTRACT_VERSION_POSITION.setStorageUint256(2**256 - 1);
+    }
+
+    /// @notice Returns the current contract version.
+    function getContractVersion() public view returns (uint256) {
```

Comment:
maybe `external`?

Reply:
Its used in _checkContractVersion and only while upgrade/init, but extra internal method will add extra byte code to Lido and NOR contracts, maybe inline getStorageUint256 in _checkContractVersion and make getContractVersion 
 external is a way?

Reply:
I see, you are right